extensions/interactions/TextInput/TextInput.js

```diff
@@ -81,3 +82,56 @@ oppia.directive('oppiaShortResponseTextInput', [
     };
   }
 ]);
+
+oppia.factory('textInputRulesService', ['$filter', function($filter) {
+  return {
+    Equals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      return normalizedAnswer.toLowerCase() == normalizedInput.toLowerCase();
+    },
+    FuzzyEquals: function(answer, inputs) {
+      var normalizedAnswer = $filter('normalizeWhitespace')(answer);
+      var answerString = normalizedAnswer.toLowerCase();
+
+      var normalizedInput = $filter('normalizeWhitespace')(inputs.x);
+      var inputString = normalizedInput.toLowerCase();
+
+      if(inputString == answerString) return true;
+      var d = [];
+      for(var i = 0; i <= inputString.length; i++){
+        d.push([i]);
+      }
+      for(var j = 0; j <= answerString.length; j++){
+        d[0].push(j);
+      }
+      for(i = 1; i <= inputString.length; i++){
```

Comment:
I don't think this loop is doing what you want it to...
If inputString = 'bcd', answerString = 'abcd', d[inputString.length][answerString.length] will be 0.

Why not just copy the implementation from normalized_string.py?